pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -828,19 +828,16 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *v1.Serv
 
 	status.Ingress = []v1.LoadBalancerIngress{{IP: loadbalancer.VipAddress}}
 
-	port, err := getPortByIP(lbaas.network, loadbalancer.VipAddress)
-	if err != nil {
-		return nil, fmt.Errorf("Error getting port for LB vip %s: %v", loadbalancer.VipAddress, err)
-	}
-	floatIP, err := getFloatingIPByPortID(lbaas.network, port.ID)
+	portID := loadbalancer.VipPortID
```

Comment:
I remember why I hate working on this code :(

https://github.com/gophercloud/gophercloud/pull/20 suggests that VipPortID will be undefined when `loadbalancer` is the result of the "create" branch way above.  Can we confirm that this actually works in the "LB doens't already exist" case on a few recent openstack releases - or perhaps change the create case way above to do a regular "get" after the "create"?